src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToCodeBehindCodeActionResolver.cs

```diff
@@ -134,18 +130,6 @@ private async Task<string> GenerateCodeBehindClassAsync(IProjectSnapshot project
 
         var newFileContent = builder.ToString();
 
-        var fixedContent = await _roslynCodeActionHelpers.GetFormattedNewFileContentsAsync(project.FilePath, codeBehindUri, newFileContent, cancellationToken).ConfigureAwait(false);
-
-        if (fixedContent is null)
-        {
-            // Sadly we can't use a "real" workspace here, because we don't have access. If we use our workspace, it wouldn't have the right settings
-            // for C# formatting, only Razor formatting, and we have no access to Roslyn's real workspace, since it could be in another process.
-            var node = await CSharpSyntaxTree.ParseText(newFileContent, cancellationToken: cancellationToken).GetRootAsync(cancellationToken).ConfigureAwait(false);
-            node = Formatter.Format(node, s_workspace, cancellationToken: cancellationToken);
-
-            return node.ToFullString();
-        }
-
-        return fixedContent;
+        return await _roslynCodeActionHelpers.GetFormattedNewFileContentsAsync(project, codeBehindUri, newFileContent, cancellationToken).ConfigureAwait(false);
```

Comment:
Can this extra async state machine be removed? Looking at `RazorCodeActionHelpers.GetFormattedNewFileContentsAsync(...)` it looks like that method returns a `Task<string>`